### PR TITLE
Add camera class

### PR DESCRIPTION
**Pull request by gabloe**

### CODE REVIEW

File: src/Asset.cpp [hunk 1]
```diff
@@ -10,7 +10,9 @@ namespace fs = std::filesystem;
 std::unique_ptr<Asset> Asset::instance;
 
 
-#if defined(__APPLE__) || defined(__MACH__) || defined(__linux__)
+#if defined(__APPLE__) || defined(__MACH__)
+uint32_t path_length = PATH_MAX;
```
Comment (deathly809) [inline src/Asset.cpp:14]: Could be a constexpr :P

File: src/TerrainGenerator.cpp [hunk 2]
```diff
@@ -167,17 +145,21 @@ void TerrainGenerator::handleKeyboardEvent(GLFWwindow* window,
...
 }
 
+void TerrainGenerator::mouseScroll(GLFWwindow* window, double xoffset, double yoffset) {
+  camera.HandleMouseScroll(static_cast<float>(yoffset));
+}
+
 void TerrainGenerator::processInput(GLFWwindow* window) {
   if (glfwGetKey(window, GLFW_KEY_W) == GLFW_PRESS) {
-    cameraPos += cameraFront * speed;
+    camera.HandleKeyboardInput(camera::MovementDirection::FORWARD, getFrameDeltaTime());
```
Comment (deathly809) [inline src/TerrainGenerator.cpp:154]: No more fast movement? :'(

File: src/TerrainGenerator.cpp [hunk 3]
```diff
@@ -126,30 +120,14 @@ void TerrainGenerator::mouseMoved(GLFWwindow* window, double x, double y) {
...
-                                           int key,
-                                           int scancode,
-                                           int action,
-                                           int mods) {
+                                            int key,
+                                            int scancode,
+                                            int action,
+                                            int mods) {
   if (mods == GLFW_MOD_SHIFT) {
     speed = running_speed;
```
Comment (deathly809) [inline src/TerrainGenerator.cpp:132]: can probable delete speed now.

File: src/Asset.cpp (hunk 1 above)
Comment (gabloe) [reply src/Asset.cpp:14]: Didn't work for me

File: src/TerrainGenerator.cpp (hunk 2 above)
Comment (gabloe) [reply src/TerrainGenerator.cpp:154]: Oh yeah, I can add it

File: src/Asset.cpp (hunk 1 above)
Comment (deathly809) [reply src/Asset.cpp:14]: Oh, that is weird. Is PATH_MAX not a constant value?

**Pull request merged**